lua/wire/stools/trigger.lua

```diff
@@ -0,0 +1,123 @@
+	language.Add( "tool.wire_trigger.desc", "Spawns a Trigger" )
+	language.Add( "Tool.wire_trigger.alltriggers", "All Triggers Visible" )
+	language.Add( "tool.wire_trigger.resetsize", "Reset Size" )
+	language.Add( "tool.wire_trigger.resetoffset", "Reset Offset" )
+	language.Add( "Tool.wire_trigger.filter_all", "All Entities" )
+	language.Add( "Tool.wire_trigger.filter_players", "Only Players" )
+	language.Add( "Tool.wire_trigger.filter_props", "Only Props" )
+	language.Add( "tool."..TOOL.Mode..".0", "Primary: Create "..TOOL.Name.."" )
+
+	concommand.Add( "wire_trigger_reset_size", function( ply, cmd, args )
+
+		RunConsoleCommand( "wire_trigger_sizex", 64 )
+		RunConsoleCommand( "wire_trigger_sizey", 64 )
+		RunConsoleCommand( "wire_trigger_sizez", 64 )
+
+	end )
+	concommand.Add( "wire_trigger_reset_offset", function( ply, cmd, args )
+
+		RunConsoleCommand( "wire_trigger_offsetx", 0 )
+		RunConsoleCommand( "wire_trigger_offsety", 0 )
+		RunConsoleCommand( "wire_trigger_offsetz", 0 )
+		
+	end )
+
+end
+function TOOL:GetConVars()
+	return self:GetClientNumber( "model" ), self:GetClientNumber( "filter" ), self:GetClientNumber( "owneronly" ), self:GetClientNumber( "sizex" ), self:GetClientNumber( "sizey" ), self:GetClientNumber( "sizez" ), self:GetClientNumber( "offsetx" ), self:GetClientNumber( "offsety" ), self:GetClientNumber( "offsetz" )
+end
+function TOOL:DrawHUD()
+	local tr = util.TraceLine( util.GetPlayerTrace( LocalPlayer() ) )
+	local ent = tr.Entity
+	if IsValid( ent ) and ent:GetClass() == "gmod_wire_trigger" or DrawOutline:GetBool() then
+
+		local list = { ent }
+		if DrawOutline:GetBool() then
+			list = ents.FindByClass( "gmod_wire_trigger" )
+		end
+		cam.Start3D( LocalPlayer():EyePos(), LocalPlayer():EyeAngles() )
+			for k,ent in pairs( list ) do
+				local trig = ent:GetTriggerEntity()
+				render.DrawWireframeBox( trig:GetPos(), Angle( 0, 0, 0 ), trig:OBBMins(), trig:OBBMaxs(), Color( 255, 255, 0 ), true )
+				render.DrawLine( trig:GetPos(), ent:GetPos(), Color( 255, 255, 0 ) )
+			end
+		cam.End3D()
+
+	end
+end
+function TOOL:RightClick( tr )
+
+	if IsValid( tr.Entity ) then
+		local ent = tr.Entity
+		if ent:GetClass() == "gmod_wire_trigger" then
+
+			-- http:--youtu.be/RTR1ny0O_io
+			local size = ent:GetTriggerSize()
+			local offset = ent:GetTriggerOffset()
+			RunConsoleCommand( "wire_trigger_sizex", size.x )
+			RunConsoleCommand( "wire_trigger_sizey", size.y )
+			RunConsoleCommand( "wire_trigger_sizez", size.z )
+			RunConsoleCommand( "wire_trigger_offsetx", offset.x )
+			RunConsoleCommand( "wire_trigger_offsety", offset.y )
+			RunConsoleCommand( "wire_trigger_offsetz", offset.z )
+			RunConsoleCommand( "wire_trigger_filter", ent:GetFilter() )
+			RunConsoleCommand( "wire_trigger_owneronly", ent:GetOwnerOnly() and 1 or 0 )
+			RunConsoleCommand( "wire_trigger_model", ent:GetModel() )
+			return true
+
+		end
+	end
+end
+
+ 
+function TOOL.BuildCPanel( panel )
```

Comment:
Would be nice to have a help text that explains what this does. See [`axis.lua`](https://github.com/garrynewman/garrysmod/blob/master/garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/axis.lua#L203).